box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/CollaborationsFragment.java

```diff
@@ -0,0 +1,322 @@
+import com.box.androidsdk.share.vm.SelectRoleShareVM;
+
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+
+public class CollaborationsFragment extends BoxFragment implements AdapterView.OnItemClickListener {
+
+    protected static final String TAG = CollaborationsFragment.class.getName();
+    private CollaboratorsAdapter mCollaboratorsAdapter;
+    private CollaborationsFragmentCallback mCallback;
+    private UsxFragmentCollaborationsBinding binding;
+    private CollaborationsShareVM mCollaborationsShareVM;
+    private SelectRoleShareVM mSelectRoleShareVM;
+
+    public interface CollaborationsFragmentCallback {
+        void notifySwitchToAccessRoleFragment();
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        setTitles();
+
+        binding = DataBindingUtil.inflate(inflater, R.layout.usx_fragment_collaborations, container, false);
+        mCollaborationsShareVM = ViewModelProviders.of(getActivity(), mShareVMFactory).get(CollaborationsShareVM.class);
+        mSelectRoleShareVM = ViewModelProviders.of(getActivity()).get(SelectRoleShareVM.class);
+        binding.setViewModel(mCollaborationsShareVM);
+        binding.setLifecycleOwner(this);
+        View view = binding.getRoot();
+        binding.collaboratorsList.setDivider(null);
+        mCollaboratorsAdapter = new CollaboratorsAdapter(getActivity(), mCollaborationsShareVM);
+        binding.collaboratorsList.setAdapter(mCollaboratorsAdapter);
+        binding.collaboratorsList.setOnItemClickListener(this);
+
+
+        mCollaborationsShareVM.getCollaborations().observe(this, onCollaborationsChange);
+        mCollaborationsShareVM.getRoleItem().observe(this, onRoleItemChange);
+        mCollaborationsShareVM.getUpdateCollaboration().observe(this ,onUpdateCollaboration);
+        mCollaborationsShareVM.getUpdateOwner().observe(this, onUpdateOwnerCollaboration);
+        mCollaborationsShareVM.getDeleteCollaboration().observe(this, onDeleteCollaboration);
+
+        mCollaborationsShareVM.getItemInfo().observe(this, onBoxItemComplete);
+        // Get serialized roles or fetch them if they are not available
+        if (getItem().getAllowedInviteeRoles() == null) {
+            fetchRoles();
+        }
+        return view;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (mSelectRoleShareVM.isRemoveSeleted()) {
+            showSpinner();
+            mCollaborationsShareVM.deleteCollaboration(mSelectRoleShareVM.getCollaboration());
+            mSelectRoleShareVM.setRemoveSelected(false); //reset remove selected
+        } else {
+            if (mSelectRoleShareVM.getSelectedRole().getValue() != null && mSelectRoleShareVM.getCollaboration() != null) {
```

Comment:
Does getSelectedRole need a null check too?

Reply:
no. it's initialized by default.